src/ext.js

```diff
@@ -65,16 +65,32 @@ export default function ext(/* env */) {
               // TODO: searchable dropdown
               type: 'string',
               ref: 'variable',
-              component: 'string',
-              label: 'Variable name',
+              component: 'dropdown',
+              defaultValue: null,
+              options: async (action, hyperCubeHandler) => {
+                const variables = await hyperCubeHandler.app.getVariableList();
+                return variables
+                  .filter(v => !v.qIsReserved || (v.qIsReserved && action.showSystemVariables))
+                  .map(v => ({
+                    label: v.qName,
+                    value: v.qName,
+                  }));
+              },
               show: data => checkShow(data, 'variable'),
             },
+            showSystemVariables: {
+              type: 'boolean',
+              ref: 'showSystemVariables',
+              label: 'showSystemVariables',
+              defaultValue: false,
+              show: data => checkShow(data, 'showSystemVariables'),
```

Comment:
Not sure if we need a specific one for showSystemVariables. We always show this when variable is selected right?

Reply:
ah true, i fix